File: oracles/Java Oracles/10 - Palindrome/Palindrome.java
public class Palindrome {
	
	public static boolean valid_char(char c) {
		if (((c >= 'A') &&
			(c <= 'Z')) ||
			((c >= 'a') &&
			(c <= 'z')))
			return true;
		else
			return false;
	}
	
	public static boolean palindrome(String str) {
		int i, j, n;
		n = str.length();
		for (i = 0; i < n; i++)
			if(!valid_char(str.charAt(i)))
				return false;
		str = str.toLowerCase();
		for (i = 0, j = n - 1; i <= j; i++, j--)
			if(str.charAt(i) != str.charAt(j))
				return false;
		return true;
	}

}

File: progtest-web/resources/oracles/C/Palindrome/Palindrome.c
#include "Palindrome.h"

unsigned char valid_char(char c) {
	if (((c >= 'A') &&
		(c <= 'Z')) ||
		((c >= 'a') &&
		(c <= 'z')))
		return 1;
	else
		return 0;
}

unsigned char palindrome(char *word) {
	unsigned char i, j, n;
	n = strlen(word);
	for (i = 0; i < n; i++)
		if(!valid_char(word[i]))
			return 0;
	for (i = 0; i < n; i++)
		if(word[i] >= 'a')
			word[i] = word[i] - ' ';
	for (i = 0, j = n - 1; i <= j; i++, j--)
		if(word[i] != word[j])
			return 0;
	return 1;
}
